src/Uintra20/Content/uintra20/src/app/ui/pages/news/uintra-news-edit/uintra-news-edit-page.component.ts

```diff
@@ -94,4 +97,17 @@ export class UintraNewsEditPage implements OnInit {
 
     return copyObject;
   }
+
+  canDeactivate(): Observable<boolean> | boolean {
+    if (this.hasDataChangedService.hasDataChanged) {
+      if(confirm('Are you sure?')) {
```

Comment:
move to service. and replace it in all places

Reply:
+